src/dataflow/src/sink/kafka.rs

```diff
@@ -339,13 +351,17 @@ where
 
                 let previously_published = match &connector.consistency {
                     Some(consistency) => match consistency.gate_ts {
-                        Some(ts) => as_of.frontier.less_equal(&ts),
+                        Some(gate_ts) => time <= gate_ts,
```

Comment:
the `less_equal` is nice to have because it generalizes nicely to multidimensional times in case we ever want to support that

Reply:
Any input on why we used the `SinkAsOf` here?

Reply:
@ruchirK Gentle ping.

Reply:
let me look again and refresh my memory! sorry for the delay

Reply:
so I think yeah using as_of was wrong here originally and using time is better, but i still think we should keep the `less_equal` function? thank you for catching this!

Reply:
I'm not sure what you mean by keep the `less_equal` function (instead of <=)? That was a function on the antichain and we're now comparing two u64s

Reply:
ah my apologies this was just a reading comprehension fail on my part :D